.github/ts-tasks/prune-github-workflow-runs.ts

```diff
@@ -0,0 +1,77 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import * as glob from "@actions/glob";
+import findRepoRoot from "find-git-root";
+import * as path from "path";
+
+const owner = "aptos-labs";
+const repo = "aptos-core";
+
+export async function pruneGithubWorkflowRuns() {
+  const githubToken = process.env.GITHUB_TOKEN;
+
+  if (!githubToken) {
+    throw new Error("Missing environment variable `GITHUB_TOKEN`");
+  }
+
+  const ghClient = github.getOctokit(githubToken);
+
+  const repoRootWithDotGit = findRepoRoot(__dirname);
+  const repoRoot = repoRootWithDotGit.substring(0, repoRootWithDotGit.length - 4); // remove the `.git` suffix from the returned path
+
+  const patterns = [`${repoRoot}/.github/worklows/*.yml`, `${repoRoot}/.github/workflows/*.yaml`];
+  const globber = await glob.create(patterns.join("\n"));
+  const workflowFilePaths = await globber.glob();
+  const workflowFilesPresentInRepo = workflowFilePaths.map((filePath) => path.basename(filePath));
+
+  if (workflowFilesPresentInRepo.length === 0) {
+    core.setFailed("Found 0 workflow files under `.github/workflows` which is kinda odd - exiting early...");
+    return;
+  }
+
+  core.info(`\nFound the following workflow files in the repo:\n${workflowFilesPresentInRepo.join("\n")}`);
+
+  const workflowResponse = await ghClient.rest.actions.listRepoWorkflows({
+    owner,
+    repo,
+  });
+
+  const obsoleteWorkflows = workflowResponse.data.workflows.filter(
+    (workflow) => !workflowFilesPresentInRepo.includes(path.basename(workflow.path)),
```

Comment:
Are workflows always named after their filenames? I thought they were overridden in the yaml spec

Reply:
no they are not, but my code uses the workflow.path property as unique identifier (the api also exposes the .path for obsolete workflows).